src/main/java/org/shanerx/tradeshop/commands/CommandCaller.java

```diff
@@ -53,14 +53,23 @@ public boolean onCommand(CommandSender sender, Command cmd, String label, String
         cmdPass = new CommandPass(sender, cmd, label, args);
         command = Commands.getType(cmdPass.getArgAt(0));
 
+
         if (!cmdPass.hasArgs()) {
             sender.sendMessage(Message.INVALID_ARGUMENTS.getPrefixed());
             return true;
 
         }
 
         if (command == null) {
-            sender.sendMessage(Message.INVALID_ARGUMENTS.getPrefixed());
+	        CustomCommandHandler handler = CustomCommandHandler.getInstance();
+	        String subcmd = cmdPass.getArgAt(0);
+
+	        if (!handler.isAvailable(subcmd)) {
+		        handler.getExecutable(subcmd).run(args);
```

Comment:
I havnt looked at #run but can we make that return a boolean and then just return that that way subCommands would be made the same as a regular command and if the maker returns false it just gives the default message.

Reply:
That is precisely how #run is intended to work.

Reply:
but then wouldn't you just be returning #run?

Reply:
is #run running the command or registering the command?

Reply:
Running it.
This is weird, I should have had `if (!...run(...)) { /* send msg for invalid args */ }`
I was actually sure I did it like that.... will fix that tomorrow.

Reply:
or you could just have it return #run and then the return is wholly on the run method

Reply:
True, but I would like every command that starts with /ts to have the same “invalid args” message. Having the addon do it leads to inconsistent behaviour, which can be baffling to the common user.

If, however, the developer of the addon truly wishes to handle usage messages they simply have to return true and they are free to output whatever they desire.
After all, the return value is not really used for anything, apart from this

Reply:
then we can catch the #run return and if false send our invalid message. I think this is the same thing you were going to do, so that sounds good.